Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift

```diff
@@ -16,6 +20,279 @@ extension Libxml2 {
+        ///     - elementDescriptor: the descriptor for the element to wrap the range in.
+        ///
+        func wrapChildren(of element: ElementNode, intersectingRange targetRange: NSRange, inElement elementDescriptor: ElementNodeDescriptor) {
+
+            let matchVerification: NodeMatchTest = { return $0 is ElementNode && elementDescriptor.matchingNames.contains($0.name) }
+
+            element.enumerateFirstDescendants(
+                in: targetRange,
+                matching: matchVerification,
+                onMatchFound: nil,
+                onMatchNotFound: { [unowned self] range in
+                    let mustFindLowestBlockLevelElements = !elementDescriptor.isBlockLevel()
+
+                    if mustFindLowestBlockLevelElements {
+                        let elementsAndIntersections = element.lowestBlockLevelElements(intersectingRange: targetRange)
+
+                        for (element, intersection) in elementsAndIntersections {
+                            // 0-length intersections are possible, but they make no sense in the context
+                            // of wrapping content inside new elements.  We should ignore zero-length
+                            // intersections.
+                            //
+                            guard intersection.length > 0 else {
+                                continue
+                            }
+
+                            self.forceWrapChildren(of: element, intersecting: intersection, inElement: elementDescriptor)
+                        }
+                    } else {
+                        self.forceWrapChildren(of: element, intersecting: targetRange, inElement: elementDescriptor)
+                    }
+            })
+        }
+
+        // MARK: - Unwrapping Nodes
+
+        /// Unwraps the specified range from nodes with the specified name.  If there are multiple
+        /// nodes with the specified name, the range will be unwrapped from all of them.
+        ///
+        /// - Parameters:
+        ///     - range: the range that must be unwrapped.
+        ///     - elementNames: the names of the elements the range must be unwrapped from.
+        ///
+        /// - Todo: this method works with node names only for now.  At some point we'll want to
+        ///         modify this to be able to do more complex lookups.  For instance we'll want
+        ///         to be able to unwrapp CSS attributes, not just nodes by name.
+        ///
+        func unwrap(range: NSRange, fromElementsNamed elementNames: [String]) {
+            unwrap(rootNode, range: range, fromElementsNamed: elementNames)
+        }
+
+        /// Unwraps the specified range from nodes with the specified name.  If there are multiple
+        /// nodes with the specified name, the range will be unwrapped from all of them.
+        ///
+        /// - Parameters:
+        ///     - range: the range that must be unwrapped.
```

Comment:
In this particular case there's no element param, but I'm adding one in the other ones. Good findings!